src/frontend/src/handler/create_source.rs

```diff
@@ -344,93 +328,58 @@ pub(crate) async fn bind_columns_from_source(
                 protobuf_schema.use_schema_registry,
             )?;
 
-            (
-                Some(
-                    extract_protobuf_table_schema(
-                        &protobuf_schema,
-                        with_properties,
-                        &mut format_encode_options_to_consume,
-                    )
-                    .await?,
-                ),
-                StreamSourceInfo {
-                    format: FormatType::Plain as i32,
-                    row_encode: EncodeType::Protobuf as i32,
-                    row_schema_location: protobuf_schema.row_schema_location.0.clone(),
-                    use_schema_registry: protobuf_schema.use_schema_registry,
-                    proto_message_name: protobuf_schema.message_name.0.clone(),
-                    key_message_name: get_key_message_name(&mut format_encode_options_to_consume),
-                    name_strategy: name_strategy
-                        .unwrap_or(PbSchemaRegistryNameStrategy::Unspecified as i32),
-                    format_encode_options,
-                    ..Default::default()
-                },
-            )
-        }
-        (Format::Plain, Encode::Json) => {
-            let schema_config = get_json_schema_location(&mut format_encode_options_to_consume)?;
-            let columns = if create_cdc_source_job {
-                Some(debezium_cdc_source_schema())
-            } else {
-                extract_json_table_schema(
-                    &schema_config,
+            stream_source_info.use_schema_registry = protobuf_schema.use_schema_registry;
+            stream_source_info.row_schema_location = protobuf_schema.row_schema_location.0.clone();
+            stream_source_info.proto_message_name = protobuf_schema.message_name.0.clone();
+            stream_source_info.key_message_name =
+                get_key_message_name(&mut format_encode_options_to_consume);
+            stream_source_info.name_strategy =
+                name_strategy.unwrap_or(PbSchemaRegistryNameStrategy::Unspecified as i32);
+
+            Some(
+                extract_protobuf_table_schema(
+                    &protobuf_schema,
```

Comment:
I believe protobuf and avro can also reuse a lot of code, however `extract_protobuf_table_schema` looks strange and I'm not brave enough to change it in this PR, which contains only equivalent rewrites. 🤡